docs/tutorials/static-types.md

```diff
@@ -0,0 +1,369 @@
+
+## Overview
+
+Static types are a way to specify the types of variables in Nextflow code, both to document the code and enable deeper forms of validation. The Nextflow language server can use type annotations to identify type-related errors during development, without needing to run the code.
+
+While Nextflow inherited type annotation from Groovy, types could only be specified for functions and local variables, and not for Nextflow-specific concepts such as processes, workflows, and pipeline parameters. Additionally, the Groovy type system is significantly larger and more complex than what is required for Nextflow pipelines.
+
+Nextflow 25.10 provides a native way to specify types at every level of a pipeline, from a pipeline parameter to a local variable in a process, using the {ref}`standard types <stdlib-types>` in the Nextflow standard library.
+
+## Developer tooling
+
+Static types are most effective when used with the [Nextflow language server](https://github.com/nextflow-io/language-server), especially in combination with the [VS Code extension](https://marketplace.visualstudio.com/items?itemName=nextflow.nextflow).
+
+### Type checking
+
+Static type checking is currently only available through the language server as an experimental feature.
+
+### Automatic migration
+
+The VS Code extension provides a command for automatically migrating Nextflow pipelines to static types. Search for and select **Convert pipeline to static types** in the command palette to migrate the current project.
+
+The language server will attempt to convert every legacy process to a typed process by migrating inputs and outputs to the new syntax. In cases where the type of an input or output cannot be inferred (e.g. `val` inputs and outputs), the type will be left unspecified, and the language server will report an error for each case.
+
+:::{tip}
+If a process has an nf-core [meta.yml](https://nf-co.re/docs/guidelines/components/modules#documentation), the language server will use it to infer the types of `val` inputs and outputs.
+:::
+
+:::{tip}
+The tooling for automatic migration to static types is actively being developed. While this page shows how to perform migrations manually, some or all of these migration steps will become automatic through developer tools such as the language server.
+:::
+
+## Example: rnaseq-nf
+
+This section demonstrates how to migrate to static types using the [rnaseq-nf](https://github.com/nextflow-io/rnaseq-nf) pipeline as an example. To view the completed migration, see the [`preview-25-10`](https://github.com/nextflow-io/rnaseq-nf/tree/preview-25-10) branch of the rnaseq-nf repository.
+
+### Initial version
+
+The [rnaseq-nf](https://github.com/nextflow-io/rnaseq-nf) pipeline performs a basic RNAseq analysis on a collection of FASTQ paired-end reads:
```

Comment:
It might be worth cutting a version of the pipeline in case master changes over time.